packages/form-js-viewer/src/Form.js

```diff
@@ -130,12 +130,30 @@ export default class Form {
    * @returns { { data: Data, errors: Errors } }
    */
   submit() {
-    this.validate();
+    const { properties = {} } = this._getState();
 
-    const {
-      data,
-      errors
-    } = this._getState();
+    const { readOnly } = properties;
+
+    if (readOnly) {

```

Comment:
Any reason we do not use the built-in `#validate` thing? Should `submit` and `validate` exibit different behaviors?

Reply:
You mean calling `validate` with some form fields or the entire form disabled?

Reply:
Should `submit`, `reset` and `validate` all throw an error if the form is disabled?

Reply:
Calling it with some fields disabled. Why don't we re-use that. What is the idea/use-case behind it. Maybe there exists one :)

Reply:
Yeah, good point. So calling `validate` with `textfield1` disabled shouldn't return errors for that form field, right?

Reply:
Validate and reset do not cause side-effects beyond the form so I'd argue no.

Submit however does (hooking up to submit is the way to trigger external submission). So I'd argue: Submit is guarded while the others are not,.

Reply:
Okay, will change that.

Reply:
Okay, so the change would be that `validate` filters disabled form fields and `submit` uses `validate` to get the errors.

Reply:
Exactly.

Reply:
Done (cf. https://github.com/bpmn-io/form-js/pull/85/commits/673e7b8a24822c4d1a3c4fb53c4546cccbd53022).